### DISCUSSION TITLE
question about the structure and data intgrity

### DISCUSSION THREAD

**Discussion by JunkyardCat** [immich-app/immich#8916] [Q&A]
so wanted to ask how immich is storing the file? can we use folder to organize?  
if we add tags or something else to help search easy will that modify the file in any way?  

does immich behave the same way as imgur ? like when i upload to imgur it changes its filename and sometimes does compress or something

**Comment by bo0tzz** [comment]
https://immich.app/docs/administration/storage-template. The original file is never modified.

**Comment by JunkyardCat** [comment]
@bo0tzz 
thanks. so it uses a template . what about tagging does it insert tags into the image? does it allow manual tagging?

**Reply by bo0tzz** [reply]
Not at the moment
